CentOS/duffy
# Add sub-command group handling DB migrations

Plus small fixes.

# How to test

- Have a Duffy configuration which points to an empty PostgreSQL(*) DB
- Create the schema
  ```
  duffy -c ... setup-db
  ```
- Fuzz with the schema
  ```
  psql ...
  alter table tenants add column foo int;
  <Ctrl+D>
  ```
- Create a migration version which realigns the database with the model (i.e. would remove the spurious column on upgrade)
  ```
  duffy -c ... migration create --autogenerate A comment describing what was changed
  ```
- Review the generated file
- Apply the migration
  ```
  duffy -c ... migration upgrade
  ```
- Verify the spurious column is gone
  ```
  psql ...
  \d tenants
  <Ctrl+D>
  ```
- Verify that Alembic knows which state the DB is in migration-wise
  ```
  duffy -c ... migration db-version
  ```
- Remove the migration file (it's bogus outside this test) and let Alembic forget about it
  ```
  rm duffy/database/migrations/versions/...
  psql ...
  delete from alembic_version;
  <Ctrl+D>
  ```

(*): SQlite is pretty featureless when it comes to altering tables, e.g. it can add columns but not remove them (you have to recreate tables with the column missing then, which is doable in alembic but way beyond what auto-generating scripts is capable of.
